atc/exec/put_step.go

```diff
@@ -59,7 +59,7 @@ func NewPutStep(
 	strategy worker.ContainerPlacementStrategy,
 	workerClient worker.Client,
 	delegate PutDelegate,
-) *PutStep {
+) Step {
```

Comment:
I'm assuming this change was unintentional from a replace command of some sort.

Reply:
No, this is a intentional change. If you look at other steps, like https://github.com/concourse/concourse/blob/master/atc/exec/get_step.go#L80, they all return `Step`, thus this is fixing a previous error. But this is also required to make the change `step_factory.go`, because, as the code:

```go
	putStep := exec.NewPutStep(
		plan.ID,
		*plan.Put,
		stepMetadata,
		containerMetadata,
		factory.resourceFactory,
		factory.resourceConfigFactory,
		factory.strategy,
		factory.client,
		delegate,
	)

	putStep = exec.LogError(putStep, delegate)
	if factory.enableRerunWhenWorkerDisappear {
		putStep = exec.RetryError(putStep, delegate)
	}
```

If `NewPutStep` returns `*PutStep`, then `putStep` is type of `*PutStep`, then we cannot step return value of `exec.LogError(putStep, delegate)` to `putStep`, because of `LogError` returns a `Step` typed object.

Reply:
Ah I see. Good catch!